docs/benchmarks/README.md

```diff
@@ -0,0 +1,151 @@
+# Benchmarking Core Operations
+
+To understand scaling and cost factors, benchmarks of independent
+operations in consistent environments is highly desireable.  The
+following document outlines the data that should be collected in
+benchmarking, generation of test data, and execution of bennchmarks for
+consistent results.
+
+## Data Collected for Comparisons
+
+From each implementation certain core pieces of metadata about the
+environment and implementation must be collected for a final benchmark
+report:
+
+- `proc` - System Processor Family and Model
+- `cores` - Number of Cores
+- `mem` - System Memory
+- `virt` - Virtualiztion Status [bare metal | containerized | vm]
+- `os` - Operating System Family
+- `osver` - Operating System Version
+- `lang` - Language of Implementation
+- `impl` - Implementation or library name
+- `alg` - The Algorithm name (e.g. p256)
+- `ver` - The version of the implementation
+- `method` - Signature method or approach [jws | jws+ld | urdna2015]
+- `contact` - Implementer contact information (typically email)
+
+The benchmarks themselves are based on aggregate data from multiple
+rounds of operations performed consistently.  Steps are taken (described
+in a later part of this document) to minimize impacts from setup and
+overhead, as well as to prevent optimization loops from skewing
+benchmark results.  The data collected from each round are as follows:
+
+- `test_id` - Unique ID of the test
+- `operation` - operation being tested
+- `round` - number of the round of executions
+- `payload_size` - size of the payload
+- `mem_start` - memory at start of round execution
+- `mem_end` - memory at end of round execution
+- `millis_start` - unix time at start of round execution
+- `millis_end` - unix time at end of round execution
+
+## Benchmark Operations
+
+There are four operations that are benchmarked as a part of this overall
+test set:
+
+- sign - actual signing of a credential (typically into a Verifiable Credential)
+- verify - verification of a signed credential
+- signPresentation - creation of a signed Verifiable Presentation
+- verifyPresentation - verification of a signed presentation
+
+## Signature Methods
+
+There are three approaches to digital signatures over credentials that
+are tested as a part of this benchmark.  Additional methods may be added
+at a later date to provide comparisons between different approaches of
+providing digital signatures of claims made regarding some subject based
+on the Verifiable Credential Core Data Model approach of signing data.
+The current methods supported are as follows:
+
+- `jws` - VCs signed and verified according to the [Verifiable
+  Credentials with JSON Web
+  Signatures](https://transmute-industries.github.io/vc-jws/)
+  specification
+- `jws+ld` - VCs signed and verified according to the [JSON Web
```

Comment:
this would be another media type... we don't currently have any registered media types for "the full verifiable credential".